numba/np/arrayobj.py

```diff
@@ -1382,6 +1382,50 @@ def impl(array, shape):
     return impl
 
 
+@register_jitable
+def numpy_broadcast_shapes_list(r, m, shape):
+    for i in range(len(shape)):
+        k = m - len(shape) + i
+        tmp = shape[i]
+        if tmp == 1:
+            continue
+        if r[k] == 1:
+            r[k] = tmp
+        elif r[k] != tmp:
+            raise ValueError("shape mismatch: objects"
+                             " cannot be broadcast"
+                             " to a single shape")
+
+
+@overload(np.broadcast_arrays)
+def numpy_broadcast_arrays(*args):
+    # number of dimensions
```

Comment:
I think this implementation needs a couple of things:
1. Some type checking to ensure that `*args` contains things that are suitable for use in the implementation.
2. Decide how much of what is accepted in NumPy's `broadcast_arrays` should be accepted by Numba. For example, the np impl accepts pretty much anything that `np.array` will accept. Should Numba do similar?

Reply:
I'll restrict the implementation to what `type_can_asarray` accepts

Reply:
Sounds good, thanks.